soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4398,7 +4433,10 @@ void RandomizerSettingsWindow::DrawElement() {
                 UIWidgets::PaddedSeparator();
 
                 // Skip Scarecrow Song
-                UIWidgets::EnhancementCheckbox(Settings::FreeScarecrow.GetName().c_str(), "gRandomizeSkipScarecrowsSong");
+                // Disabled when Shuffle Ocarina Buttons is active
```

Comment:
I don't think this needs to be disabled with shuffle buttons, you simply remove the button check in logic and in the SkipScarecrow check if this is enabled. 
FreeScarecrow is the setting var for it in logic, you can use it as a condition like any other item. Look at BombchusInLogic to see what I mean.
In the SkipScarecrow check, you should be able to use Randomizer_GetSettingValue to check if scarecrow is free.

Reply:
Fixed